mautic/mautic
# Report improvements with contact attribution and dashboard widget support

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | Y
| New feature?  | Y
| BC breaks?    |  N
| Deprecations? | N
| Fixed issues  |  

## Description

This PR is dependent on #1713 and #1770. Both will need to be merged BEFORE this one. 

This PR does several things:

1. Applies the date filter to tabular data in addition to graphs which should help stabilize the reports for large data sets
2. Adds new contact attribution support that tracks touch points in a single location and also reporting/graphs to go along with it.
3. Adds support to choose any report's graph as a dashboard widget which also displays a button/link to the full report
4. Provides greater flexibility to bundles to build reports
5. Adds option to choose which filters should be "dynamic" which displays the filter in the new filter shelf (little tab beside the details tab on the report details view)
6. Random stuff like removing postgres lingering code
7. Fixes a bug where data is not included for the date to

## Steps to reproduce bug

1. Hit up a page to generate some page hits in the database for today.
2. Change the date range to be from yesterday to today
3. View a page hits graph and notice it does not show data
4. This is because the to date is set to be midnight (start) of the day and not end of day so it excludes todays data.

## Steps to test this PR

1. Apply the PR and clear symfony's cache
2. Run the `doctrine:migrations:migrate` script
3. Go to reports and create a few new reports and play around
4. Edit a lead and assign the Attribution field a number (dollar amount) and an attribution date
5. Execute a campaign and follow up with decisions for the lead in #4 (open an email, visit a page, etc).
6. Create the new Attribution reports and review the data
7. Create a new widget for the dashboard and choose a reports graph. Click the view full report button to see the full report. 
8. Dynamically filter a report by clicking the new Filters shelf that will contain dynamic filters. Enter something and hit enter to filter the report. These filters will override those set in the report itself. Clearing the dynamic filters will result in the default filters being applied.

For the bug,
Check the graphs to see that the hour goes up to the current hour and is includes today's hits.
